### PR TITLE
Bump controller-runtime to 0.17 / kube to 1.29

### PR DESCRIPTION
**Pull request by justinsb**
- chore: Update to controller-runtime 0.17 / kube 1.29
- deprecation: Use NewDynamicRESTMapper instead of NewDiscoveryRESTMapper

### CODE REVIEW

**Comment by tomasaschan** [comment]
Let me know if I can help speed this along somehow!

**Comment by atoato88** [comment]
This PR is related with the content of #379.
I think #379 can be closed if this PR is merged.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/kubebuilder-declarative-pattern
 
-go 1.20
+go 1.21
+
+toolchain go1.22.0
```
Comment (atoato88) [inline go.mod:5]: Maybe, we should update the files for GitHub Actions?
- https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/.github/workflows/main.yml
This file use the go v1.20 setup at 2 place.

**Review by atoato88 — COMMENTED** [review]
I know this PR is WIP, but I left just one comment.

**Comment by atoato88** [comment]
@justinsb 
Ping, how about the current state of this?

File: go.mod (hunk 1 above)
Comment (justinsb) [reply go.mod:5]: You're right - I just pushed that and tests are looking much happier!

Comment (justinsb) [reply go.mod:5]: And I just saw that @tomasaschan proposed a better version of the change in https://github.com/justinsb/kubebuilder-declarative-pattern/pull/1 , so using that one

**Comment by justinsb** [comment]
Removed WIP, and I also think we should consider expanding ownership of this repo, perhaps starting with @tomasaschan .  Sometimes I get pulled into different things and things fall off my plate - sorry for the delays here!

**Comment by tomasaschan** [comment]
/lgtm

(but I suspect it won't matter yet...)

**Pull request merged**